### DISCUSSION TITLE
Tracing which back-ends provided the completion

### DISCUSSION THREAD

**Discussion by fehmud** [company-mode/company-mode#1168] [Q&A] [answered]
When Company does not complete as expected, it is difficult to determine why that happened. If Company provided an option to log in the *Messages* buffer a message like _Company completed with `<back-end>'_, then that would help with troubleshooting.

For instance, `company-irony` -- a third-party back-end -- overlaps with `company-clang` that comes with Company [1]. Therefore `company-irony` must come before `company-clang` in `company-backends`, or it won't work. Even when `company-irony` comes first, it could still fail to provide completions because of misconfiguration, but then it would be still useful to know that `company-clang' was called instead. 

I would have provided a pull request for this, but I couldn't determine where Company selected the back-ends.  If you could provide some guidance, I will submit a pull request.

Thanks for your attention.
## 

[1] https://github.com/Sarcasm/company-irony/issues/2

**Comment by dgutov** [comment, marked as answer]
You can type `M-x company-diag`, and the diagnostic buffer will show which backend was used.

**Comment by fehmud** [comment]
Oh!  You had implemented this already.  Thank you.
